cloudformation/xtdb_ecs.yml

```diff
@@ -0,0 +1,239 @@
+      ClusterName: !Ref ClusterName
+      
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: 'xtdb-standalone'
+      RetentionInDays: 365
+
+  TaskExecutionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service: 'ecs-tasks.amazonaws.com'
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
+
+  ECSServiceRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ecs.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: ecs-service
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'elasticloadbalancing:DeregisterInstancesFromLoadBalancer'
+                  - 'elasticloadbalancing:DeregisterTargets'
+                  - 'elasticloadbalancing:Describe*'
+                  - 'elasticloadbalancing:RegisterInstancesWithLoadBalancer'
+                  - 'elasticloadbalancing:RegisterTargets'
+                  - 'ec2:Describe*'
+                  - 'ec2:AuthorizeSecurityGroupIngress'
+                Resource: '*'
+
+  TaskDefinition:
+    Type: 'AWS::ECS::TaskDefinition'
+    Properties:
+      Family: 'xtdb-task'
+      RequiresCompatibilities:
+        - 'EC2'
+      ExecutionRoleArn: !GetAtt TaskExecutionRole.Arn
+      NetworkMode: 'bridge'
+      ContainerDefinitions:
+        - Name: 'xtdb-container'
+          Image: 'ghcr.io/xtdb/xtdb-aws-ea:latest'
+          Essential: true
+          PortMappings:
+            - ContainerPort: 3000
+          MemoryReservation: 256 
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-group: !Ref LogGroup
+              awslogs-region: !Ref 'AWS::Region'
+              awslogs-stream-prefix: xtdb
+          Environment:
+            - Name: KAFKA_BOOTSTRAP_SERVERS
+              Value: !Ref MSKBootstrapServers
+            - Name: XTDB_TOPIC_NAME
+              Value: xtdb-log
+            - Name: XTDB_S3_BUCKET
+              Value: !Ref S3BucketName
+            - Name: XTDB_SNS_TOPIC
+              Value: !Ref SNSTopicArn
+
+  EC2InstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument: 
+        Version: "2012-10-17"
+        Statement:
+          - Sid: ''
+            Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role'
+        - !Ref S3AccessPolicyArn
```

Comment:
can/should these go on the ECS service role rather than the machine itself?

Reply:
I don't think they belong in the ECS service role - my understanding is that is more for the ECS service to register container instances. I did try placing them in the `TaskExecutionRole` - but that didn't work, it says the following when attempting to set up s3:

```
"User: arn:aws:sts::<accountid>:assumed-role/xtdb-stack-ecs-EC2InstanceRole-cq4KLVh142Ez/i-0221dfee3da46d752 is not authorized to perform: sqs:createqueue on resource: arn:aws:sqs:eu-west-1:<account-id>:xtdb-object-store-notifs-queue-7de3c104-82b7-4463-a224-d93f831776d7 because no identity-based policy allows the sqs:createqueue action (Service: Sqs, Status Code: 403, Request ID: fa4ced72-32cf-51b5-9c7f-b0923e8b76df, Extended Request ID: null)"}}
```

Seems to use the instance role for these things, so planning to leave as is?

Reply:
that's weird, I could have sworn AWS gave you a way to give permissions to the individual service/task - they're normally pretty fine grained when it comes to these things, I doubt they'd want to give any service running on that machine the same permissions, and it'd also mean that the permissions of the service depended on the machine it happens to run on?

Reply:
Ahhhh - that was it! Got you - yeah, does sound like a **task role** is the way forward. I'll look about bringing that in and testing all reboots and works correctly :)